src/test/java/org/broadinstitute/barclay/help/DocGenBaseTest.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.barclay.help;
+
+import com.sun.javadoc.ClassDoc;
+import com.sun.javadoc.Tag;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class DocGenBaseTest {
+
+    protected final static HelpDoclet HELP_DOCLET = new HelpDoclet();
+    protected final static DefaultDocWorkUnitHandler DEFAULT_DOC_WORK_UNIT_HANDLER =
+            new DefaultDocWorkUnitHandler(HELP_DOCLET);
+
+    /**
+     * Generates an empty ClassDoc (mocked).
+     */
+    protected ClassDoc emptyClassDoc() {
+        return mockClassDoc("");
+    }
+
+    /**
+     * Generates an ClassDoc with the javadoc text
+     */
+    protected ClassDoc mockClassDoc(final String javadocText) {
+        return mockClassDoc(javadocText, Collections.emptyMap());
+    }
+
+
+    /**
+     * @param inlineTags map of (custom) tag names in javadoc to its text.
+     *
+     * @return mocked class doc.
+     */
+    protected ClassDoc mockClassDoc(final String javadocText,
+            final Map<String, String> inlineTags) {
+        // mock class
+        final ClassDoc mockedClassDoc = Mockito.mock(ClassDoc.class);
+        // mock the javadoc text
+        final Tag[] javadoc = Arrays.stream(javadocText.split("\n"))
+                .map(line -> mockTag(null, line))
+                .toArray(Tag[]::new);
+        Mockito.when(mockedClassDoc.firstSentenceTags()).thenReturn(javadoc);
+        final Tag[] inline = inlineTags.entrySet().stream()
+                .map(entry -> mockTag(entry.getKey(), entry.getValue()))
+                .toArray(Tag[]::new);
+        Mockito.when(mockedClassDoc.inlineTags()).thenReturn(inline);
+        return mockedClassDoc;
+    }
+
+
+    private Tag mockTag(final String name, final String text) {
+        final Tag tag = Mockito.mock(Tag.class);
+        Mockito.when(tag.name()).thenReturn(name);
+        Mockito.when(tag.text()).thenReturn(text);
+        return tag;
+    }
+
+    protected DocWorkUnit createDocWorkUnit(final Class<?> clazz, final ClassDoc classDoc) {
+        return new DocWorkUnit(
+                DEFAULT_DOC_WORK_UNIT_HANDLER,
+                clazz.getAnnotation(DocumentedFeature.class),
+                classDoc,
+                clazz
+        );
+    }
+
+    protected DocWorkUnit createDocWorkUnit(final Class<?> clazz) {
+        return createDocWorkUnit(clazz, emptyClassDoc());
+    }
+}
```

Comment:
I think it might be more useful for this class to something that is instantiated by unit test classes, rather than be a superclass for them. Then the tests could instantiate one of these (maybe passing in whatever doclet or workunit class they want), or just call static methods, and create appropriate mocks for the given test. This would make the tests totally self-contained (i.e., the testPropertyMap test method below would have everything it depends on defined locally, right in the method), and this class could then be used to write unit tests for, say, both the default doclet and the "test" doclet. At some point, we might even want to create another test doclet class to test some feature, and that doclet could also be passed to this class to use to create mocks. Just looking at this it doesn't seem that hard to do, and would be more flexible, but let me know if I'm missing something.